### PR TITLE
Use user profile timezone from DB as fallback for CalDAV all-day events

### LINKED ISSUE #8754: [CAL-1631] Incorrect blocked times in CalDAV all-day events
The current implementation doesn't handle timezone information properly for CalDAV all-day events. This leads to incorrect timezone handling for these events, causing potential scheduling conflicts and user confusion. 

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-1631](https://linear.app/calcom/issue/CAL-1631/incorrect-blocked-times-in-caldav-all-day-events)</sub>

### PR DESCRIPTION
**Pull request by alishaz-polymath**
## What does this PR do?

The iCalendar 2.0 specification does not require timezone information for all-day events. All-day events are represented by a date value instead of a date-time value, which means they don't include timezone information. An all-day event is assumed to occur at the same local time across all time zones.

This PR addresses the issue where the tzid is not always present in CalDAV all-day events, as per the iCalendar 2.0 specification. To ensure proper timezone handling, we have made the following changes:

1. Added a new function getUserId to safely retrieve the user ID from the selectedCalendars array.
2. Fetched the userTimeZone from the user's profile using the obtained user ID.
3. Modified the existing code to use the userTimeZone as a fallback when the tzid is missing from the vcalendar events.

By implementing these changes, we ensure that the timezone information is accurately reflected in CalDAV all-day events, even when the timezone data is not explicitly provided within the event itself.

Fixes #8754 

**Environment**: Staging(main branch)

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

## How should this be tested?

- [ ] Please test an All-Day event with CalDAV connected calendar. The slots should be blocked for the day that's supposed to have the all day event if the Timezone information within the Cal.com profile is the same as the booking page. This is essentially because, as mentioned above, CalDAV all day events **do not** send timezone information as part of the event details, which means we must fallback to their cal.com profile to fetch nearest accurate timezone information.

### CODE REVIEW

**Review by alishaz-polymath — COMMENTED** [review]
Self review provided

File: packages/lib/CalendarService.ts [hunk 1]
```diff
@@ -276,6 +276,38 @@ export default abstract class BaseCalendarService implements Calendar {
...
+    const user = await prisma.user.findUnique({
+      where: {
+        id,
+      },
+      select: {
+        timeZone: true,
+      },
+    });
+    return user?.timeZone || "Europe/London";
+  };
```
[the remark this answers is not in the document]
Comment (alishaz-polymath) [reply packages/lib/CalendarService.ts:296]: Oh yes!
Forgot to change the return type. Good eyes man! 👀

File: packages/lib/CalendarService.ts [hunk 2]
```diff
@@ -276,6 +276,38 @@ export default abstract class BaseCalendarService implements Calendar {
...
+    const prisma = await import("@calcom/prisma").then((mod) => mod.default);
+    const user = await prisma.user.findUnique({
+      where: {
+        id,
+      },
+      select: {
+        timeZone: true,
+      },
+    });
+    return user?.timeZone || "Europe/London";
```
Comment (emrysal) [inline packages/lib/CalendarService.ts:295]: Please don't default the tz here; it'd make debugging harder.

**Review by emrysal — APPROVED** [review]
LGTM - Looks good to me

**Pull request merged**